tests/console/kmod.pm

```diff
@@ -0,0 +1,71 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test common commands: modinfo, lsmod, modprobe, rmmod,
+# depmod. Tests the commands and their output for common words that
+# should appear independently of the individual modules loaded in
+# each system.
+# -modinfo: The command is called, exit status is asserted, and then
+# output is checked for expected words (like 'filename ... .ko',
+# 'license ... GPL', 'author ... Qumranet' - output that should
+# be present in every system)
+# -lsmod: The command is called and it's output is checked for the
+# words that should appear at the top after successful excecution:
+# Module | Size | Used by.
+# -modprobe: rmmod is first called to make sure arc4 module is not
+# active, and then modeprobe is called and checks for the appropriate
+# output when the module starts.
+# -rmmod: rmmod is called and exit status is checked, and then
+# modprobe enables the module again to make sure it was stopped
+# by rmmod in the first place.
+# -depmod: the command is called, it's exit code is checked, and then
+# the output is checked for words that should appear in every system,
+# like 'lib/modules', '.ko' and 'kernel'.
+# Maintainer: Vasilios Anastasiadis <vasilios.anastasiadis@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+
+sub run {
+    select_console 'root-console';
+
+    #test 1
+    #test modinfo command
+    assert_script_run('OUT="$(modinfo kvm)"');
+    #test the output for common words that should always appear
+    assert_script_run('grep -o -m 1 \'^filename:.*kvm.ko\' <<< "$OUT"');
+    assert_script_run('grep -o -m 1 \'^license:.*GPL\' <<< "$OUT"');
+    assert_script_run('grep -o -m 1 \'^author:.*Qumranet\' <<< "$OUT"');
+    assert_script_run('grep -o -m 1 \'^parm:.*ignore_msrs:bool\' <<< "$OUT"');
+    #test 2
+    #test lsmod command
+    #test that the output has the expected correct format
+    assert_script_run('lsmod | grep \'^Module.*Size.*Used by$\'');
+    #test 3
+    #test modprobe -v arc4 command
+    script_run('rmmod arc4');    #make sure the module is not loaded
```

Comment:
If you really want to make sure use `-f` ?
On line 59 you use `assert_script_run`, any reason to use one or another way to run it ?

Reply:
Hi Josef, thanks for the input!
In line 55 I use script_run since I don't care about the exit code, I just wanted to make sure it's not loaded and rmmod it if it was, so I can test modprobe right after. 

Then, in line 59 I am actually testing rmmod so I care about whether it was successful or not (I know it is loaded because I used modprobe right before). That's why I assert this time. Then I run modprobe one last time to make sure that rmmod did the work.

Reply:
good, thank you